BierBuddy.UILib/FindBuddiesPageRenderer.cs

```diff
@@ -0,0 +1,460 @@
+            gridFactory.AppendChild(contentPresenterFactory);
+
+            template.VisualTree = gridFactory;
+
+            return template;
+        }
+        private UIElement GetProfileBorder()
+        {
+            Border profileBorder = new();
+            profileBorder.Background = UIUtils.Outer_Space;
+            profileBorder.CornerRadius = UIUtils.UniversalCornerRadius;
+            profileBorder.Child = profilePanel;
+            return profileBorder;
+        }
+        private void SetProfilePanel(double width, double height)
+        {
+            profilePanel.Width = width;
+            profilePanel.Height = height - 150;
+            UIElement profilePicture = GetProfilePicture(width);
+            profilePanel.Children.Add(profilePicture);
+
+            UIElement profileContent = GetProfileContentPanel(width);
+            Canvas.SetTop(profileContent, height - UIUtils.ProfileConentHeight - 150);
+            profilePanel.Children.Add(profileContent);
+        }
+        
+        private UIElement GetProfilePicture(double width)
+        {
+            Label tempFoto = new Label();
+            tempFoto.Content = "FOTO!";
+            tempFoto.Foreground = UIUtils.testMarking;
+            return  tempFoto;
+        }
+
+        private UIElement GetProfileContentPanel(double width)
+        {
+            // Create the Grid
+            Grid profileGrid = new Grid();
+            profileGrid.Width = width;
+            profileGrid.Height = UIUtils.ProfileConentHeight;
+            profileGrid.HorizontalAlignment = HorizontalAlignment.Left;
+            profileGrid.VerticalAlignment = VerticalAlignment.Top;
+
+            // Define the Columns
+            ColumnDefinition colDef1 = new ColumnDefinition();
+            ColumnDefinition colDef2 = new ColumnDefinition();
+            ColumnDefinition colDef3 = new ColumnDefinition();
+            profileGrid.ColumnDefinitions.Add(colDef1);
+            profileGrid.ColumnDefinitions.Add(colDef2);
+            profileGrid.ColumnDefinitions.Add(colDef3);
+
+            // Define the Rows
+            RowDefinition rowDef1 = new RowDefinition();
+            RowDefinition rowDef2 = new RowDefinition();
+            RowDefinition rowDef3 = new RowDefinition();
+            RowDefinition rowDef4 = new RowDefinition();
+            rowDef4.Height = new GridLength(UIUtils.ProfileConentHeight / 6);
+            profileGrid.RowDefinitions.Add(rowDef1);
+            profileGrid.RowDefinitions.Add(rowDef2);
+            profileGrid.RowDefinitions.Add(rowDef3);
+            profileGrid.RowDefinitions.Add(rowDef4);
+
+            UIElement ProfileBanner = GetProfileBanner();
+            Grid.SetColumnSpan(ProfileBanner, 3);
+            Grid.SetRow(ProfileBanner, 1);
+
+            UIElement drinkLabel;
+            UIElement activityLabel;
+            UIElement interestLabel;
+
```

Comment:
De preferences en interesses kunnen niet 0 zijn, in de database moet er minstens 1 aanwezig zijn van elk, anders breekt de database zelf al.